src/core/pointcloud/qgscopcpointcloudindex.h

```diff
@@ -84,11 +84,11 @@ class CORE_EXPORT QgsCopcPointCloudIndex: public QgsPointCloudIndex
     void copyCommonProperties( QgsCopcPointCloudIndex *destination ) const;
 
     /**
-     * Returns the gps time flag from global_encoding field in LAS header, 0 indicates GPS week time (seconds passed since the beginning of the week)
+     * Returns one datapoint, "CopcGpsTimeFlag": The gps time flag from global_encoding field in LAS header,
+     * 0 indicates GPS week time (seconds passed since the beginning of the week)
      * 1 indicates GPS adjusted time, which is seconds passed since the GPS base time minus 1e9
-     * \since QGIS 3.42
      */
-    bool gpsTimeFlag() const;
+    QVariantMap extraMetadata() const override;
```

Comment:
This is more fragile vs the explicit getter -- I'd rather not see us use QVariantMap for purposes like this

Reply:
@nyalldawson the reasoning is that the gpsTimeFlag is something very specific to one particular file format, with quite minor impact, and it does not deserve to be a part of the public API. And it seemed that over time, we may want to pass more format-specific metadata read from the file header - hence introduction of this fairly generic extraMetadata() function. Given that it is not used for any core functionality, it seems okay to use QVariantMap in this case (I also don't like to see QVariantMap in API)